src/libraries/System.Private.CoreLib/src/System/TimeSpan.cs

```diff
@@ -103,6 +164,24 @@ public TimeSpan(int days, int hours, int minutes, int seconds, int milliseconds)
 
         public int Milliseconds => (int)((_ticks / TicksPerMillisecond) % 1000);
 
+        /// <summary>
+        /// Gets the microseconds component of the time interval represented by the current <see cref="TimeSpan"/> structure.
+        /// </summary>
+        /// <remarks>
+        /// The <see cref="Microseconds"/> property represents whole microseconds, whereas the
+        /// <see cref="TotalMicroseconds"/> property represents whole and fractional microseconds.
+        /// </remarks>
+        public int Microseconds => (int)((_ticks / TicksPerMicrosecond) % 1000);
+
+        /// <summary>
+        /// Gets the nanoseconds component of the time interval represented by the current <see cref="TimeSpan"/> structure.
+        /// </summary>
+        /// <remarks>
+        /// The <see cref="Nanoseconds"/> property represents whole nanoseconds, whereas the
+        /// <see cref="TotalNanoseconds"/> property represents whole and fractional nanoseconds.
+        /// </remarks>
+        public int Nanoseconds => (int)((_ticks * NanosecondsPerTick) % 1000);
```

Comment:
This looks wrong. 

```C#
public int Nanoseconds => (int)((_ticks % TicksPerMicrosecond) * 100;
```

Reply:
Why `TicksPerMicrosecond`? Its Nanosecond

Reply:
I am getting the number if ticks of the fraction of the Microseconds. Note, I am using `%` operator.  This fraction should reflect the nanoseconds after multiplying it by 100.